cg/models/demultiplex/run_parameters.py

```diff
@@ -20,61 +19,70 @@ def __init__(self, run_parameters_path: Path):
         self.tree: ElementTree = read_xml(file_path=run_parameters_path)
         self.validate_instrument()
 
+    def validate_instrument(self) -> None:
+        """Raise an error if the parent class was instantiated."""
+        raise NotImplementedError(
+            "Parent class instantiated. Instantiate instead RunParametersHiSeq, "
+            "RunParametersNovaSeq6000 or RunParametersNovaSeqX"
+        )
+
+    def is_single_index(self) -> bool:
+        """Return False if the sequencing is not HiSeq. Overriden in HiSeq"""
+        return False
+
     @property
     def index_length(self) -> int:
         """Return the length of the indexes if they are equal, raise an error otherwise."""
         index_one_length: int = self.get_index_1_cycles()
         index_two_length: int = self.get_index_2_cycles()
-        if index_one_length != index_two_length:
+        if index_one_length != index_two_length and not self.is_single_index():
             raise RunParametersError("Index lengths are not the same!")
         return index_one_length
 
-    @staticmethod
-    def node_not_found(node: Optional[ElementTree.Element], name: str) -> None:
-        """Raise exception if the given node is not found."""
-        if node is None:
-            message = f"Could not determine {name}"
-            LOG.warning(message)
-            raise RunParametersError(message)
+    def get_tree_node(self, node_name: str, name: str) -> ElementTree.Element:
+        """Return the node of a tree given its name if it exists."""
+        xml_node = self.tree.find(node_name)
+        validate_node_exists(node=xml_node, name=name)
+        return xml_node
+
+    def get_node_string_value(self, node_name: str, name: str) -> str:
+        """Return the value of the node as a string if its validation passes."""
+        return self.get_tree_node(node_name=node_name, name=name).text
 
-    def validate_instrument(self) -> None:
-        """Raise an error if the parent class was instantiated."""
-        raise NotImplementedError(
-            "Parent class instantiated. Instantiate instead RunParametersNovaSeq6000 or RunParametersNovaSeqX"
-        )
+    def get_node_integer_value(self, node_name: str, name: str) -> int:
+        """Return the value of the node as an integer if its validation passes."""
+        return int(self.get_node_string_value(node_name=node_name, name=name))
 
     @property
-    def control_software_version(self) -> Optional[str]:
+    def control_software_version(self) -> str | None:
```

Comment:
I'm guessing `RunParameters` is an abstract class? This is usually how it is written in Python, if you want to be more idiomatic.

```python
from abc import ABC, abstractmethod

class MyAbstractClass(ABC):

    @abstractmethod
    def abstract_method(self):
        """This method should be overridden by any non-abstract subclass."""
        pass

    def concrete_method(self):
        """This is a concrete method with its own implementation."""
        print("This is a concrete method.")

# Subclassing the abstract base class
class Subclass(MyAbstractClass):

    def abstract_method(self):
        """Implements the abstract method from the abstract base class."""
        print("Implementation of the abstract method.")

```

Reply:
Yes, for some reason pycharm complains when you put pass in these methods

Reply:
in the case of properties, it says `Getter should return or yield something`

Reply:
Oh, I totally missed the abstract method decorator on top, implementing it solves the warning 🙂